GPU/Common/VertexShaderGenerator.cpp

```diff
@@ -1277,7 +1277,11 @@ bool GenerateVertexShader(const VShaderID &id, char *buffer, const ShaderLanguag
 		}
 
 		// Compute fogdepth
-		WRITE(p, "  %sv_fogdepth = (viewPos.z + u_fogcoef.x) * u_fogcoef.y;\n", compat.vsOutPrefix);
+		WRITE(p, "  if (u_fogcoef.x == -1.0 && u_fogcoef.y == -1.0) {\n");
```

Comment:
I'm not convinced these aren't values a game could use.  If viewPos.z was in range [-1, 1] it could be sensible to subtract 1 and flip the sign.

If both are `>= 65535.0`, that would seem pretty safe.

-[Unknown]